pyomo/contrib/solver/config.py

```diff
@@ -0,0 +1,377 @@
+            description=description,
+            doc=doc,
+            implicit=implicit,
+            implicit_domain=implicit_domain,
+            visibility=visibility,
+        )
+
+        self.tee: bool = self.declare(
+            'tee',
+            ConfigValue(
+                domain=bool,
+                default=False,
+                description="If True, the solver log prints to stdout.",
+            ),
+        )
+        self.log_solver_output: bool = self.declare(
+            'log_solver_output',
+            ConfigValue(
+                domain=bool,
+                default=False,
+                description="If True, the solver output gets logged.",
+            ),
+        )
+        self.working_dir: str = self.declare(
+            'working_dir',
+            ConfigValue(
+                domain=str,
+                default=None,
+                description="The directory in which generated files should be saved. "
+                "This replaces the `keepfiles` option.",
+            ),
+        )
+        self.load_solutions: bool = self.declare(
+            'load_solutions',
+            ConfigValue(
+                domain=bool,
+                default=True,
+                description="If True, the values of the primal variables will be loaded into the model.",
+            ),
+        )
+        self.raise_exception_on_nonoptimal_result: bool = self.declare(
+            'raise_exception_on_nonoptimal_result',
+            ConfigValue(
+                domain=bool,
+                default=True,
+                description="If False, the `solve` method will continue processing "
+                "even if the returned result is nonoptimal.",
+            ),
+        )
+        self.symbolic_solver_labels: bool = self.declare(
+            'symbolic_solver_labels',
+            ConfigValue(
+                domain=bool,
+                default=False,
+                description="If True, the names given to the solver will reflect the names of the Pyomo components. "
+                "Cannot be changed after set_instance is called.",
+            ),
+        )
+        self.timer: HierarchicalTimer = self.declare(
+            'timer', ConfigValue(default=None, description="A HierarchicalTimer.")
+        )
+        self.threads: Optional[int] = self.declare(
+            'threads',
+            ConfigValue(
+                domain=NonNegativeInt,
+                description="Number of threads to be used by a solver.",
+                default=None,
+            ),
+        )
+        self.time_limit: Optional[float] = self.declare(
+            'time_limit',
+            ConfigValue(
+                domain=NonNegativeFloat, description="Time limit applied to the solver."
+            ),
+        )
+        self.solver_options: ConfigDict = self.declare(
+            'solver_options',
+            ConfigDict(implicit=True, description="Options to pass to the solver."),
+        )
```

Comment:
- Do you have a suggestion?
- Let's tackle that when we start moving over more solvers. I don't want to over-engineer in advance.

Reply:
I think long ago we discussed this being a mapping of options to solvers for things like PyROS that rely on multiple subsolvers, which is an idea I kind of like. It would be nice if we did the magic for there being only one set of options (like here) under the hood though, as PyROS and friends are a pretty niche use case. I don't hate the name `solver_options` (I'm generally much more offended by calling the solver an algorithm than calling it a solver :P), but historically we've just called this `options`, and I've never though that was unclear?